storage/src/checker/mod.rs

```diff
@@ -150,14 +153,23 @@ impl<S: WritableStorage> NodeStore<Committed, S> {
 
         // read the node and iterate over the children if branch node
         let node = self.read_node(subtrie_root_address)?;
+        let mut current_node_path = path_prefix.clone();
+        current_node_path.0.extend_from_slice(node.partial_path());
+        #[cfg(feature = "ethhash")]
```

Comment:
While we're here, if the nibble length is odd, there can't be a value either, regardless of ethhash.